rust_src/src/lread.rs

```diff
@@ -404,4 +405,52 @@ pub fn read_char_exclusive(
     }
 }
 
+/// Delete the symbol named NAME, if any, from OBARRAY.
+/// The value is t if a symbol was found and deleted, nil otherwise.
+/// NAME may be a string or a symbol.  If it is a symbol, that symbol
+/// is deleted, if it belongs to OBARRAY--no other symbol is deleted.
+/// OBARRAY, if nil, defaults to the value of the variable `obarray'.
+/// usage: (unintern NAME OBARRAY)
+#[lisp_fn(min = "1")]
+pub fn unintern(name: LispSymbolOrString, obarray: Option<LispObarrayRef>) -> LispObject {
+    let obarray = obarray.unwrap_or_else(LispObarrayRef::global);
+    let obarray: LispObarrayRef = check_obarray(obarray.into()).into();
```

Comment:
Let's make a `check` method on `LispObarrayRef` and have the `check_obarray` function call it.

Reply:
Sure, makes sense. I wasn't sure what to do with the corresponding doc comments, so I left them as-is, with `check_obarray`. Let me know if these should be moved to `check` instead.